src/Agent.Listener/Configuration/WindowsSecurityManager.cs

```diff
@@ -0,0 +1,477 @@
+                }
+            }
+
+            return exists;
+        }
+
+        public void CreateLocalGroup(string groupName)
+        {
+            Trace.Info(nameof(CreateLocalGroup));
+            LocalGroupInfo groupInfo = new LocalGroupInfo();
+            groupInfo.Name = groupName;
+            groupInfo.Comment = StringUtil.Format("Built-in group used by Team Foundation Server.");
+
+            int returnCode = NetLocalGroupAdd(null, 1, ref groupInfo, 0);
+
+            // return on success
+            if (returnCode == ReturnCode.S_OK)
+            {
+                return;
+            }
+
+            // Error Cases
+            switch (returnCode)
+            {
+                case ReturnCode.NERR_GroupExists:
+                case ReturnCode.ERROR_ALIAS_EXISTS:
+                    Trace.Info(StringUtil.Format("Group {0} already exists", groupName));
+                    break;
+                case ReturnCode.ERROR_ACCESS_DENIED:
+                    throw new UnauthorizedAccessException(StringUtil.Format("Access Denied."));
+
+                case ReturnCode.ERROR_INVALID_PARAMETER:
+                    throw new ArgumentException(StringUtil.Format("Invalid Group Name - {0}.", groupName));
+
+                default:
+                    throw new Exception(StringUtil.Format("Error: Operation {0} with Return Code {1}.", "NetLocalGroupAdd", returnCode.ToString()));
+            }
+        }
+
+        public void AddMemberToLocalGroup(string accountName, string groupName)
+        {
+            Trace.Info(nameof(AddMemberToLocalGroup));
+            LocalGroupMemberInfo memberInfo = new LocalGroupMemberInfo();
+            memberInfo.FullName = accountName;
+
+            int returnCode = NetLocalGroupAddMembers(null, groupName, 3, ref memberInfo, 1);
+
+            // return on success
+            if (returnCode == ReturnCode.S_OK)
+            {
+                return;
+            }
+
+            // Error Cases
+            switch (returnCode)
+            {
+                case ReturnCode.ERROR_MEMBER_IN_ALIAS:
+                    Trace.Info(StringUtil.Format("Account {0} is already member of group {1}", accountName, groupName));
+                    break;
+                case ReturnCode.NERR_GroupNotFound:
+                case ReturnCode.ERROR_NO_SUCH_ALIAS:
+                    throw new ArgumentException(StringUtil.Format("Group: {0} Not Exist.", groupName));
+
+                case ReturnCode.ERROR_NO_SUCH_MEMBER:
+                    throw new ArgumentException(StringUtil.Format("Member: {0} Not Exist.", accountName));
+
+                case ReturnCode.ERROR_INVALID_MEMBER:
+                    throw new ArgumentException(StringUtil.Format("A new member could not be added to a local group because the member has the wrong account type. If you are configuring on a domain controller, built-in machine accounts cannot be added to local groups. You must use a domain user account instead."));
```

Comment:
None of the errors from this file are using facing.. up above there is a comment i have kept which mentions that.. The caller captures any exception thrown here, logs it and prints generic error that it couldn't grant permission to the user. In this file there are lot of strings, IMO, I dont think all of them need to be localized. Just the required.